ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HivePointLookupOptimizerRule.java

```diff
@@ -599,7 +587,7 @@ public ConstraintGroup(RexNode rexNode) {
           if (constraint == null) {
             throw new SemanticException("Unable to find constraint which was earlier added.");
           }
-          ret.add(constraint.exprNode);
+          ret.add(constraint.constNode);
```

Comment:
oh my; Constrain had it's constructor arguments swapped! what a typo!

Reply:
I was also surprised we did not catch it before! But that's OK, it was working as expected and now they are swapped back!